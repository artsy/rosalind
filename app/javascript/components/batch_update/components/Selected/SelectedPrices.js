import React from 'react'
import SelectedComponent from './SelectedComponent'

const SelectedPrices = props => {
  const { minPrice, maxPrice, clearState } = props
  return (
    <div>
      <h2>Price</h2>
      {minPrice !== null && (
        <SelectedComponent
          className="currentMinPrice"
          stateKey="minPrice"
          name={'Minimum: USD ' + minPrice}
          onRemove={clearState}
        />
      )}
      {maxPrice !== null && (
        <SelectedComponent
          className="currentMaxPrice"
          stateKey="maxPrice"
          name={'Maximum: USD ' + maxPrice}
          onRemove={clearState}
        />
      )}
    </div>
  )
}

export { SelectedPrices }
