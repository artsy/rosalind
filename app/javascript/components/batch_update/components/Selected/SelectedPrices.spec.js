import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { SelectedPrices } from './SelectedPrices'

let props

beforeEach(() => {
  props = {
    minPrice: 1000,
    maxPrice: 2000,
    clearState: jest.fn(),
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedPrices {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when only min price is set', () => {
  props.maxPrice = null
  const rendered = renderer.create(<SelectedPrices {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when only max price is set', () => {
  props.minPrice = null
  const rendered = renderer.create(<SelectedPrices {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
