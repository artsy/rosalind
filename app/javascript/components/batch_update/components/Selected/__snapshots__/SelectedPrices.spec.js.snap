// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  font-weight: bold;
  margin: 0.25em 0;
}

.c0 a.remove {
  display: inline-block;
  height: 2em;
  width: 2em;
  line-height: 2em;
  text-align: center;
  color: #ddd;
  margin-left: 0.5em;
  border-radius: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a.remove:hover {
  color: #333;
  cursor: pointer;
  background: #eee;
}

<div>
  <h2>
    Price
  </h2>
  <div
    className="currentMinPrice c0"
  >
    Minimum: USD 1000
    <a
      className="remove"
      href="#"
      onClick={[Function]}
    >
      ✕
    </a>
  </div>
  <div
    className="currentMaxPrice c0"
  >
    Maximum: USD 2000
    <a
      className="remove"
      href="#"
      onClick={[Function]}
    >
      ✕
    </a>
  </div>
</div>
`;

exports[`renders correctly when only max price is set 1`] = `
.c0 {
  font-weight: bold;
  margin: 0.25em 0;
}

.c0 a.remove {
  display: inline-block;
  height: 2em;
  width: 2em;
  line-height: 2em;
  text-align: center;
  color: #ddd;
  margin-left: 0.5em;
  border-radius: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a.remove:hover {
  color: #333;
  cursor: pointer;
  background: #eee;
}

<div>
  <h2>
    Price
  </h2>
  <div
    className="currentMaxPrice c0"
  >
    Maximum: USD 2000
    <a
      className="remove"
      href="#"
      onClick={[Function]}
    >
      ✕
    </a>
  </div>
</div>
`;

exports[`renders correctly when only min price is set 1`] = `
.c0 {
  font-weight: bold;
  margin: 0.25em 0;
}

.c0 a.remove {
  display: inline-block;
  height: 2em;
  width: 2em;
  line-height: 2em;
  text-align: center;
  color: #ddd;
  margin-left: 0.5em;
  border-radius: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a.remove:hover {
  color: #333;
  cursor: pointer;
  background: #eee;
}

<div>
  <h2>
    Price
  </h2>
  <div
    className="currentMinPrice c0"
  >
    Minimum: USD 1000
    <a
      className="remove"
      href="#"
      onClick={[Function]}
    >
      ✕
    </a>
  </div>
</div>
`;
